packages/pigeon/lib/java_generator.dart

```diff
@@ -606,46 +606,50 @@ class JavaGenerator extends StructuredGenerator<JavaOptions> {
-                  // a Dart 'int'.  To keep things simple we just use 64bit
-                  // longs in Pigeon with Java.
-                  final bool isInt = arg.type.baseName == 'int';
-                  final String argType =
-                      isInt ? 'Number' : _javaTypeForDartType(arg.type);
-                  final String argName = _getSafeArgumentName(index, arg);
-                  final String argExpression = isInt
-                      ? '($argName == null) ? null : $argName.longValue()'
-                      : argName;
-                  String accessor = 'args.get($index)';
-                  if (isEnum(root, arg.type)) {
-                    accessor = _intToEnum(accessor, arg.type.baseName);
-                  } else if (argType != 'Object') {
-                    accessor = '($argType) $accessor';
-                  }
-                  indent.writeln('$argType $argName = $accessor;');
-                  if (!arg.type.isNullable) {
-                    indent.write('if ($argName == null) ');
-                    indent.addScoped('{', '}', () {
-                      indent.writeln(
-                          'throw new NullPointerException("$argName unexpectedly null.");');
-                    });
-                  }
-                  methodArgument.add(argExpression);
-                });
-              }
-              if (method.isAsynchronous) {
-                final String resultValue =
-                    method.returnType.isVoid ? 'null' : 'result';
-                const String resultName = 'resultCallback';
-                indent.format('''
+                    'reply.reply(wrapError(new IllegalArgumentException("Arguments expected but none received.")));');
+                indent.writeln('return;');
+              });
+              enumerate(method.arguments, (int index, NamedType arg) {
+                // The StandardMessageCodec can give us [Integer, Long] for
+                // a Dart 'int'.  To keep things simple we just use 64bit
+                // longs in Pigeon with Java.
+                final bool isInt = arg.type.baseName == 'int';
+                final String argType =
+                    isInt ? 'Number' : _javaTypeForDartType(arg.type);
+                final String argName = _getSafeArgumentName(index, arg);
+                final String argExpression = isInt
+                    ? '($argName == null) ? null : $argName.longValue()'
+                    : argName;
+                String accessor = 'args.get($index)';
+                if (isEnum(root, arg.type)) {
+                  accessor = _intToEnum(accessor, arg.type.baseName);
+                } else if (argType != 'Object') {
+                  accessor = '($argType) $accessor';
+                }
+                indent.writeln('$argType $argName = $accessor;');
+                if (!arg.type.isNullable) {
+                  indent.write('if ($argName == null) ');
```

Comment:
Similarly, we can just remove this.

Reply:
https://github.com/flutter/packages/blob/b6668e4ef35f3f18373e43df2d699e3da8091dfd/packages/pigeon/platform_tests/alternate_language_test_plugin/android/src/test/java/com/example/alternate_language_test_plugin/PigeonTest.java#L30-L32

This test expects a `NullPointerException` when a argument is unexpectedly null, I assume it should also be removed? Otherwise it fails without the null check.

Reply:
Yes, the test should be removed since we're deliberately removing the behavior.